tests/unit/test_article_upload.py

```diff
@@ -1,12 +1,29 @@
 import json
+from os import path
+
+from invenio_workflows import Workflow
+from workflow.engine_db import WorkflowStatus
 
-from scoap3.modules.compliance.models import Compliance
 from scoap3.modules.records.util import create_from_json
+from tests.responses import get_response_dir
+
+
+def common(input_json_filename):
```

Comment:
I like the idea of putting common things in a shared "fixture", but, in principle, it should only run the logic itself. You should leave the assertions to the tests and let them fail explicitly there.

I would structure it in a slightly different way:
- in the test
     open file
     pass data to util function
     get result from util function and assert something
- util function
  just run some logic and return some values

Maybe also rename `common` in something more meaningful like `run_workflow`?

Reply:
So basically you say that the "common" function should only contain the workflow creation and execution from the parsed json? I see why this is good, but in this case all test would contain the same copy-paste file reading part.

Reply:
it won't be copy/paste but just calling a function and asserting the result, it's ok :) we'll gain a lot in terms of readability